iac-extensions/arm/src/main/java/org/sonar/iac/arm/tree/impl/bicep/FileImpl.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * SonarQube IaC Plugin
+ * Copyright (C) 2021-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.arm.tree.impl.bicep;
+
+import java.util.ArrayList;
+import java.util.List;
+import javax.annotation.CheckForNull;
+import org.sonar.iac.arm.tree.api.File;
+import org.sonar.iac.arm.tree.api.Statement;
+import org.sonar.iac.arm.tree.api.StringLiteral;
+import org.sonar.iac.arm.tree.api.bicep.SyntaxToken;
+import org.sonar.iac.arm.tree.impl.AbstractArmTreeImpl;
+import org.sonar.iac.common.api.tree.Tree;
+
+public class FileImpl extends AbstractArmTreeImpl implements File {
```

Comment:
I have some mixed feelings. This duplicates a lot of what is done for json. However, I see the point of having a dedicated Bicep implementation here. An abstract class of FileImpl does not necessarily means it is more readable.

Reply:
I'm facing the same situation on my other ticket, with very similar implementation of `NumericLiteral`, `BooleanLiteral` and `NullLiteral`.
And yes, while I was tempted to have a common abstract class, I was also afraid that it would add a layer that would reduce readability.

Reply:
Another option is to move some of the method implementations to the interface. Especially the getKind() method. So `default getKind() {...}`

Reply:
I like this approach, I will give it a try in my next ticket!